src/main/java/org/folio/service/InterfaceService.java

```diff
@@ -1,79 +1,76 @@
 package org.folio.service;
 
 import static javax.ws.rs.core.Response.Status.NOT_FOUND;
-import static org.folio.util.ResponseUtils.handleFailure;
-import static org.folio.util.ResponseUtils.handleNoContentResponse;
 
 import javax.ws.rs.core.Response;
 
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.folio.persist.CriterionBuilder;
-import org.folio.persist.Tx;
+import org.folio.rest.persist.Conn;
 import org.folio.rest.persist.PostgresClient;
 import org.folio.rest.persist.Criteria.Criterion;
+import org.folio.util.ResponseUtils;
 
 import io.vertx.core.AsyncResult;
-import io.vertx.core.Context;
 import io.vertx.core.Future;
 import io.vertx.core.Handler;
 import io.vertx.core.Promise;
 import io.vertx.ext.web.handler.HttpException;
 
 public class InterfaceService {
 
+  private static final Logger logger = LogManager.getLogger(InterfaceService.class);
+
   private static final String INTERFACE_CREDENTIALS_TABLE = "interface_credentials";
   private static final String INTERFACE_TABLE = "interfaces";
 
-  private final Logger logger = LogManager.getLogger(this.getClass());
   private final PostgresClient pgClient;
 
   public InterfaceService(PostgresClient pgClient) {
     this.pgClient = pgClient;
   }
 
-  public void deleteOrganizationsInterfaceById(String id,
-    Context vertxContext, Handler<AsyncResult<Response>> asyncResultHandler) {
-    vertxContext.runOnContext(v -> {
-      Tx<String> tx = new Tx<>(id, pgClient);
-      tx.startTx()
-        .compose(this::deleteCredentialByInterfaceId)
-        .compose(this::deleteInterfaceById)
-        .compose(Tx::endTx)
-        .onComplete(handleNoContentResponse(asyncResultHandler, tx, "Interface '{}' '{}' deleted"));
-    });
+  public void deleteOrganizationsInterfaceById(String id, Handler<AsyncResult<Response>> asyncResultHandler) {
+    pgClient.withTrans(
+      conn ->
+        deleteCredentialByInterfaceId(conn, id)
+          .compose(conn1 -> deleteInterfaceById(conn1, id))
+          .onComplete(result -> ResponseUtils.handleNoContentResponse(result, id, "Interface '{}' '{}' deleted", asyncResultHandler))
+    );
   }
 
-  private Future<Tx<String>> deleteCredentialByInterfaceId(Tx<String> tx) {
-    logger.debug("Trying to delete credential by interfaceId: {}", tx.getEntity());
+  private Future<Conn> deleteCredentialByInterfaceId(Conn conn, String id) {
+    logger.debug("Trying to delete credential by interfaceId: {}", id);
 
-    Promise<Tx<String>> promise = Promise.promise();
+    Promise<Conn> promise = Promise.promise();
     Criterion criterion = new CriterionBuilder()
-      .with("interfaceId", tx.getEntity()).build();
-    pgClient.delete(tx.getConnection(), INTERFACE_CREDENTIALS_TABLE, criterion, reply -> {
+      .with("interfaceId", id).build();
+    pgClient.delete(INTERFACE_CREDENTIALS_TABLE, criterion, reply -> {
```

Comment:
pgClient.delete will create or get from the pool new sql connection, and will not use Conn that you passed to the method.
So proposed change: pgClient.delete() -> conn.delete()